File: typescript/bad_examples/naive_datetime/naive_datetime.ts
/**
 * Deliberate read of the ambient clock, which the lint gate forbids.
 *
 * `new Date()` without arguments hides a non-deterministic dependency;
 * time must be injected explicitly. The ban is a custom no-restricted-syntax
 * selector documented in eslint.config.js and LANG_SPEC.md. See
 * bad_examples/README.md for the manifest of expected signals.
 */
export function auditStamp(): Date {
  return new Date();
}

File: go/bad_examples/naive_datetime/naive_datetime.go
// Package naive_datetime reads the wall clock directly instead of going
// through the injected Clock port; forbidigo bans time.Now outside
// internal/adapters/clock.go, so the lint gate must reject this file.
//
// See bad_examples/README.md for the manifest of expected signals.
package naive_datetime

import "time"

// SnapshotAt hides a nondeterministic wall-clock read behind a pure-looking
// name.
func SnapshotAt() time.Time {
	return time.Now()
}
